docs/csharp/pattern-matching.md

```diff
@@ -2,31 +2,225 @@
 title: Pattern Matching | C# Guide
 description: Learn about pattern matching expressions in C#
 keywords: .NET, .NET Core, C#
-ms.date: 10/04/2016
+ms.date: 01/24/2017
+ms.author: wiwagn
 ms.topic: article
 ms.prod: .net
 ms.technology: devlang-csharp
 ms.devlang: csharp
 ms.assetid: 1e575c32-2e2b-4425-9dca-7d118f3ed15b
 ---
 
-# 🔧 Pattern Matching
+# Pattern Matching #
 
-> **Note**
-> 
-> This topic hasn’t been written yet! 
->
-> We welcome your input to help shape the scope and approach. You can track the status and provide input on this
-> [issue](https://github.com/dotnet/docs/issues/1114) at GitHub.
-> 
-> If you would like to review early drafts and outlines of this topic, please leave a note with your contact information in the issue.
->
-> Learn more about how you can contribute on [GitHub](https://github.com/dotnet/docs/blob/master/CONTRIBUTING.md).
->
+Patterns test that a value has a certain *shape*, and can *extract*
+information from the value when it has the matching shape. Pattern
+matching provides more concise syntax for algorithms you already use
+today. You already create pattern matching algorithms using existing
+syntax. You write `if` or `switch` statements that test values. Then,
+when those statements match, you extract and use information from that
+value. The new syntax elements are extensions to statements you are already
+familiar with: `is` and `switch`. These new extensions combine testing
+a value and extracting that information.
 
-<!--
+In this topic, we'll look at the new syntax to show you how it enables
+readable, concise code. Pattern matching enables idioms you'll need more
+often building distributed systems, where data and the methods that manipulate
+that data are separate. 
 
-<< Build the sample in an additive way >>
-<< Do the shape thing for the https://docs.microsoft.com/en-us/dotnet/articles/fsharp/language-reference/discriminated-unions shapes >>
+To illustrate these new idioms, let's work with structures that represent
+geometric shapes using pattern matching statements. You are probably
+familiar with building class hierarchies and creating
+[virtual methods and overridden methods](methods.md#inherited) to
+customize object behavior based on the runtime type of the object.
+
+Those techniques aren't possible for data that isn't structured in a class
+hierarchy. When data and methods are separate, you need other tools. The new
+*pattern matching* constructs enable cleaner syntax to examine data
+and manipulate control flow based on any condition of that data.
+
+In this topic, you'll build a method  that computes the area of
+different geometric shapes. But, you'll do it without resorting to object
+oriented techniques and building a class hierarchy for the different shapes.
+You'll use *pattern matching* instead. To further emphasize that we're not
+using inheritance, you'll make each shape a `struct` instead of a class. 
```

Comment:
For context, it might be good to add that structs can't explicitly inherit from another type.